### ISSUE TITLE
signature for s3 does not work

### ISSUE THREAD

**Issue by shotaspost** [mhart/aws4#5]
Signature for s3 does not work. Aws gives me "The request signature we calculated does not match the signature you provided. Check your key and signing method."

The cause seems to be that this module signs "s3.ap-northeast-1.amazonaws.com", whereas it should sign "s3-ap-northeast-1.amazonaws.com"

**Comment by shotaspost** [comment]
Sorry, I mean

The cause seems to be that this module signs "s3.ap-northeast-1.amazonaws.com", whereas it should sign "s3-ap-northeast-1.amazonaws.com"

**Comment by mhart** [comment]
Ah, s3 is a bit annoying like that. 👿

I'll look into this tomorrow – thanks for the report!

**Comment by mhart** [comment]
So I'm assuming you're just using the `region` parameter?

It works currently just fine if you supply the host:

```js
var opts = aws4.sign({service: 's3', region: 'ap-northeast-1', host: 's3-ap-northeast-1.amazonaws.com'})
```

**Comment by mhart** [comment]
Fixed in v1.0.3 – let me know if that works for you.

**Comment by shotaspost** [comment]
It worked! Thanks for the good work!
